### DISCUSSION TITLE
Some transforms export as XForms, some don't

### DISCUSSION THREAD

**Discussion by macevhicz** [Autodesk/maya-usd#3940] [Q&A] [answered]
We have a rig that when animated and exported to USD, only exports **some** of the Maya transform nodes as Usd XForms, which appear in the usd file like:
```
def Xform "Geo" (
    kind = "group"
)
```

But for some reason, some of the transforms instead are defined like this in the usd file:
```
def "left_body" (
)
```

I tried adding the `Usd_kind` attr, per the docs, which results in:
```
def "left_body" (
    kind = "group"
)
```

In both of those last two cases, they are not defined as `XForm`, and therefore they have no tranlate, rotate, etc values, and the result sits at origin.

I've compared the maya attributes (as seen in the Attribute Editor) between a transform that exports as XForm, and one that doesn't and cannot see any relevant differences. 

Any ideas why this may be? Or perhaps guidance in where I might look in the code for where the exporter determines which type of prim to define?

Maya2024 
MayaUsdPlugin 0.25.0

**Comment by macevhicz** [comment, marked as answer]
Welp, this issue can be ignored. We recieved the model from another vendor, and somewhere along the way, various transforms had the `Usd_typeName` attr added, but renamed to `typeName` so they weren't obvious as Usd attrs, and they had no value. Removing these fixed the issue.

**Comment by wallworm** [comment]
Thanks for sharing and updating the issue.
